sync/trypush.py

```diff
@@ -450,6 +451,71 @@ def delete(self):
                 idx.delete(key, data)
 
 
+class TcPush(TryPush):
```

Comment:
So overall, having seen it, this feels a little weird. The other classes that we store as top-level objects are things that where we're (at least conceptually) in control of the lifecycle. But in this case we aren't in control and just have a single point of interaction with the taskcluster run, so the only thing that we do with this object is call `download_logs` and it's otherwise exposing a bunch on unused API surface, at least some of which doens't make sense.

I think we should just keep the `download_logs` function, but otherwise remove this class. If we want to store metadata about which taskgroups we've previously processes, it might make sense to have `upstream_tc_taskgroup` on the sync object, for example.

Reply:
I agree, it does feel out of place in this use case. Going to remove this change, and have a function in `trypush.py` for downloading the logs.